test/utils/Arrays.test.js

```diff
@@ -0,0 +1,81 @@
+const ArraysImpl = artifacts.require('ArraysImpl');
+
+const BigNumber = web3.BigNumber;
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+contract('Arrays', function () {
+  context('Odd number of elements', function () {
```

Comment:
Love the testing for odd and even elements! We should also add a test for an empty array (and decide what to do in that scenario).